src/main/java/net/dv8tion/jda/api/entities/User.java

```diff
@@ -225,4 +228,171 @@ default String getEffectiveAvatarUrl()
      */
     @Nonnull
     JDA getJDA();
+
+    /**
+     * Returns the {@link net.dv8tion.jda.api.entities.User.UserFlag UserFlags} of this user.
+     * 
+     * @return EnumSet containing the flags of the user.
+     */
+    @Nonnull
+    EnumSet<UserFlag> getFlags();
+
+    /**
+     * Represents the bit offsets used by Discord for public flags
+     */
+    enum UserFlag
+    {
+        STAFF(                     0, "Discord Employee"),
+        PARTNER(                   1, "Discord Partner"),
+        HYPESQUAD(                 2, "HypeSquad Events"),
+        BUG_HUNTER_LEVEL_1(        3, "Bug Hunter Level 1"),
+
+        // HypeSquad
+        HYPESQUAD_BRAVERY(         6, "HypeSquad Bravery"),
+        HYPESQUAD_BRILLIANCE(      7, "HypeSquad Brilliance"),
+        HYPESQUAD_BALANCE(         8, "HypeSquad Balance"),
+
+        EARLY_SUPPORTER(           9, "Early Supporter"),
+        TEAM_USER(                10, "Team User"),
+        SYSTEM(                   12, "System User"),
+        BUG_HUNTER_LEVEL_2(       14, "Bug Hunter Level 2"),
+        VERIFIED_BOT(             16, "Verified Bot"),
+        VERIFIED_DEVELOPER(       17, "Verified Bot Developer"),
+        
+        UNKNOWN(                   -1, "Unknown");
+
+        /**
+         * Empty array of UserFlag enum, useful for optimized use in {@link java.util.Collection#toArray(Object[])}.
+         */
+        public static final UserFlag[] EMPTY_FLAGS = new UserFlag[0];
+        
+        private final int offset;
+        private final int raw;
+        private final String name;
+
+        UserFlag(int offset, @Nonnull String name)
+        {
+            this.offset = offset;
+            this.raw = 1 << offset;
+            this.name = name;
+        }
+
+        /**
+         * The readable name as used in the Discord Client.
+         * 
+         * @return The readable name of this {@link net.dv8tion.jda.api.entities.User.UserFlag UserFlag}.
```

Comment:
Linking to the same class is redundant

Reply:
This should then be corrected in the Permission enum too.
https://github.com/DV8FromTheWorld/JDA/blob/cc11f87957c2512ce33f84664df9df7ad43e2908/src/main/java/net/dv8tion/jda/api/Permission.java#L129
https://github.com/DV8FromTheWorld/JDA/blob/cc11f87957c2512ce33f84664df9df7ad43e2908/src/main/java/net/dv8tion/jda/api/Permission.java#L142
https://github.com/DV8FromTheWorld/JDA/blob/cc11f87957c2512ce33f84664df9df7ad43e2908/src/main/java/net/dv8tion/jda/api/Permission.java#L203-L204
https://github.com/DV8FromTheWorld/JDA/blob/cc11f87957c2512ce33f84664df9df7ad43e2908/src/main/java/net/dv8tion/jda/api/Permission.java#L208
https://github.com/DV8FromTheWorld/JDA/blob/cc11f87957c2512ce33f84664df9df7ad43e2908/src/main/java/net/dv8tion/jda/api/Permission.java#L224
https://github.com/DV8FromTheWorld/JDA/blob/cc11f87957c2512ce33f84664df9df7ad43e2908/src/main/java/net/dv8tion/jda/api/Permission.java#L233
https://github.com/DV8FromTheWorld/JDA/blob/cc11f87957c2512ce33f84664df9df7ad43e2908/src/main/java/net/dv8tion/jda/api/Permission.java#L251
https://github.com/DV8FromTheWorld/JDA/blob/cc11f87957c2512ce33f84664df9df7ad43e2908/src/main/java/net/dv8tion/jda/api/Permission.java#L272

Reply:
The permission enum is not the topic of this PR.

Reply:
I did remove any links to the same class in the UserFlag enum